### PR TITLE
Rework block index and size

### PR DESCRIPTION
**Pull request by sam-golioth**

### CODE REVIEW

File: src/coap_blockwise.c [hunk 1]
```diff
@@ -80,24 +81,27 @@ static void on_block_sent(struct golioth_client *client,
...
     int err = ctx->callback.read_cb(ctx->block_idx,
                                     ctx->block_buffer,
-                                    &ctx->block_size,
+                                    &block_size,
                                     &ctx->is_last,
                                     ctx->callback_arg);
     if (err != GOLIOTH_OK)
     {
-        // TODO: handle application callback error
+        return -1 * err;
```
Comment (mniestroj) [inline src/coap_blockwise.c:94]: why not `return -err;`?

File: src/coap_blockwise.c [hunk 2]
```diff
@@ -137,11 +141,14 @@ static enum golioth_status process_blockwise_uploads(struct golioth_client *clie
...
 {
     enum golioth_status status = GOLIOTH_ERR_FAIL;
-    ctx->block_size = CONFIG_GOLIOTH_BLOCKWISE_UPLOAD_BLOCK_SIZE;
-    call_read_block_callback(ctx);
-    if (ctx->block_size)
+    int ret = call_read_block_callback(ctx);
+    if (ret < 0)
     {
-        status = upload_single_block(client, ctx);
+        status = -1 * ret;
```
Comment (mniestroj) [inline src/coap_blockwise.c:147]: Same here. Curious about this convention. I haven't seen it before.

**Review by mniestroj — APPROVED** [review]
LGTM, with just minor comments.

File: src/coap_blockwise.c (hunk 1 above)
Comment (sam-golioth) [reply src/coap_blockwise.c:94]: Honestly I was just working on this too late last night

**Pull request merged**